docs/geode-native-docs/serialization/data-serialization.html.md.erb

```diff
@@ -27,3 +27,249 @@ the best combination of versatility and ease-of-use for most applications.
+- You must provide serialization instructions for objects of this class. Specifically, you must implement the `ToData()` and `FromData()` methods for .NET applications, or the `toData()` and `fromData()` methods in C++ applications.
+- The class must be registered with the cache, which takes care of informing the server.
+
+The examples define the serializable class, Orders, and supplies its serialization and deserialization methods.
+Once these pieces are in place, execution is simple: the main.cpp routine registers the serializable class then performs some put and get operations.
+
+## Execution
+
+Both examples perform the same sequence of operations, displaying simple log entries as they run.
+
+- To run an example, follow the instructions in the README.md file in the example directory.
+- Review the source code in the example directory to see exactly how it operates.
+
+- Begin by running a script that sets up the server-side environment by invoking `gfsh` commands to create a region.
+
+- Run the example client application, which performs the following steps:
+
+  - Connects to the server
+  - Registers the PdxSerializable class
+  - Creates orders
+  - Stores orders
+  - Retrieves orders
+
+
+### .NET Example
+
+This section contains code snippets showing highlights of the .NET PdxSerializable example. They are not intended for cut-and-paste execution.
+For the complete source, see the example source directory.
+
+The .NET example defines a PdxSerializable class called `Order` that inherits from the `IPdxSerializable` interface.
+An `Order` object contains three fields:
+
+- an integer `order_id`
+- a string `name`
+- a short-int `quantity`
+
+From Order.cs:
+
+```csharp
+  public class Order : IPdxSerializable
+  {
+    ...
+    public long OrderId { get; set; }
+    public string Name { get; set; }
+    public short Quantity { get; set; }
+```
+
+Using the `IPdxSerializable` read and write methods, the `Order` class defines `ToData()` and `FromData()`
+methods that perform the serialization and deserialization operations, respectively.
+
+From Order.cs:
+
+```csharp
+    public void ToData(IPdxWriter output)
+    {
+      output.WriteLong(ORDER_ID_KEY_, OrderId);
+      output.MarkIdentityField(ORDER_ID_KEY_);
+
+      output.WriteString(NAME_KEY_, Name);
+      output.MarkIdentityField(NAME_KEY_);
+
+      output.WriteInt(QUANTITY_KEY_, Quantity);
+      output.MarkIdentityField(QUANTITY_KEY_);
+    }
+
+    public void FromData(IPdxReader input)
+    {
+      OrderId = input.ReadLong(ORDER_ID_KEY_);
+      Name = input.ReadString(NAME_KEY_);
+      Quantity = (short)input.ReadInt(QUANTITY_KEY_);
+    }
+```
+
+The .NET example mainline creates a cache, then uses it to register the PdxSerializable class that was created in Orders.cs:
+
+```csharp
+   var cacheFactory = new CacheFactory()
+       .Set("log-level", "none");
+   var cache = cacheFactory.Create();
+
+   cache.TypeRegistry.RegisterPdxType(Order.CreateDeserializable);
```

Comment:
The factory method CreateDeserializable() is called on the Order class in the registration. I thought we had this method noted in the "what's required for pdxserializable" section, in the past.

Reply:
It doesn't matter what the method is named, but that it returns an instance of the object. I will try to look through older documentation.

Reply:
Also, a default constructor must be provided and used by the factory method that will return the instance of the custom object.

Reply:
Did not add mention of default constructor, as it's specific to autoregistration on the .NET side. Did add mentions of factory class and code snippets showing its definition and use.